codegen/writers/endpoint.go

```diff
@@ -29,19 +32,80 @@ type (
 	}
 )
 
+type endpointWriter struct {
+	sections   []*codegen.Section
+	outputPath string
+}
+
+func (e endpointWriter) Sections() []*codegen.Section {
+	return e.sections
+}
+
+func (e endpointWriter) OutputPath() string {
+	return e.outputPath
+}
+
 // EndpointsWriter returns the codegen.FileWriter for the endpoints of the given
 // service.
 func EndpointsWriter(api *design.APIExpr, service *design.ServiceExpr) codegen.FileWriter {
-	return nil
+	return codegen.FileWriter(endpointWriter{
+		sections: []*codegen.Section{
+			codegen.Header("", "endpoints", []*codegen.ImportSpec{
+				&codegen.ImportSpec{Path: "context"},
+				&codegen.ImportSpec{Path: "goa.design/goa.v2"},
+				&codegen.ImportSpec{Path: "goa.design/goa.v2/examples/account/gen/services"},
+			}),
+			&codegen.Section{
+				Template: *endpointTmpl,
+				Data:     buildEndpointData(api, service),
+			},
+		},
+		outputPath: "gen/endpoints/", // TODO Set output path.
+	})
 }
 
-// endpointT is the template used to write a endpoint definition.
-const endpointT = `
-type (
-	{{ comment .Description 1 }}
-	{{ .Name }} interface {{{ range .Methods }}
-		{{ comment .Description 2 }}
-		{{ .Name }}(context.Context, {{ gotypename .Payload }}) ({{ gotypename .Result }}, error)
+func buildEndpointData(api *design.APIExpr, service *design.ServiceExpr) endpointData {
+	methods := make([]*endpointMethod, len(service.Endpoints))
+	for i, v := range service.Endpoints {
+		methods[i] = &endpointMethod{
+			Name:        v.Name,
+			Description: v.Description,
+		}
+		if payload, ok := v.Payload.(*design.UserTypeExpr); ok {
+			methods[i].Payload = payload
+		}
+		if result, ok := v.Result.(*design.UserTypeExpr); ok {
```

Comment:
A result type could also be a media type. At this level we don't need to differentiate so you can use the `design.UserType` interface I believe.